NitroxClient/Communication/ClientBridge.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using NitroxClient.Communication.Abstract;
+using NitroxModel.Packets;
+using NitroxClient.Communication.Exceptions;
+using System.Collections.Generic;
+using NitroxModel.PlayerSlot;
+using NitroxModel.Logger;
+
+namespace NitroxClient.Communication
+{
+    //This is meant to be a drop-in replacement for the old PacketSender. It was re-written in accordance with the Bridge design pattern to clarify intent.
+    //This pattern should give us greater flexibility in how we bridge the connection between the server and the client moving forward. 
+    //Example: We can refactor this implementation into a ReservableClientBridge, allowing us to implement different types of client bridges depending on specific requirements.
+    public class ClientBridge : IClientBridge
```

Comment:
Does this class name leak implementation details of bridge pattern?  Since we can just remove packet sender in the same PR should we stick with that naming?  Thoughts?

Reply:
It is a placeholder for lack of a better name in the moment =) 

If I were to rename it, then I would lean towards SessionManager or something along those lines as it is lower in the graph than the packet sender abstraction is. The intent of this class is to negotiate a session with the client and ensure that standard packet transmission does not occur out-of-process, whereas a PacketSender passes packets to the client unless it has been turned off.

Honestly, I'm hesitant to move on a rename without a chance to envision where we are going to go with our netcode in general. I feel like the process for negotiating a session is going to vary between normal and privileged users, but I don't know that for sure - its just a gut feeling. If not, then the intent of this layer narrows just a bit. 

The intent at the moment is to set us up with a bridge. However, we may end up pairing this down to a facade or something along those lines if it turns out that this isn't going to be as complex as I'm anticipating.

Reply:
In the future we might want to make it possible to connect with different protocols, simultaneously. But that's mostly a server-side issue, if we were to support a TCP/UDP connection to it while also accepting another player who connects through the Steam network (and potentially another player in another way).

To go even further, being able to run the server locally inside the game (similar to how minecraft does it) has always been an idea floating around in my mind. Though this has some severe complications, such that we can't break the existing game up in a server-part dealing with physics and game logic, and a client-part doing the rendering, input interaction and communication to the server. I haven't yet thought of a way to make it both functional and efficient at the same time.